### DISCUSSION TITLE
Vehicle should be moved to domain-specific spec

### DISCUSSION THREAD

**Discussion by pkapustin** [beckn/protocol-specifications#254] [Core Specification]
Vehicle isn't needed in core and should be moved to the domain-specific taxi spec.

**Comment by techframewirk** [comment]
I would like to hear the thoughts on how important it is having the `Vehicle` object be uniform across domains. For example if the object is used in 2 completely separate domains, will it be beneficial that they both use the same structure? One pro would be that the data model will be uniform across domains. But would the meaning and fields radically change from domain to domain that it would not be useful at all?

I believe if someone is trying to use the specification on a domain that was not thought of or implemented so far, they will usually be using the core spec as the reference. So having the object there would ensure that this new domain version of the spec will be uniform to the core as well as other domain specs as much as possible. But this can leads to the question whether all new objects that are used in domain specifications be available as a central collection of objects that anyone can refer from in the core spec. Should there be a decision based on the new object to have it in the core spec if it seems that the object will useful in other domains?

**Comment by ravi-prakash-v** [comment]
Vehicle may seem like a mobility/logistics specific entity but it is an important entity that captures the movement of goods from one place to another. It is an integral part of commerce that avoids a lot of interoperability issues. I am okay abstracting it to something else like "Carrier" or something as long as it refers to an entity that carries the "Payload" of a "Fulfillment". If the carrier has a moving location, it could be constructed as a Vehicle and if it is static, it could become a "Room" or "Storage". But I do not think it is a good idea to remove it completely. Please abstract this entity and then remove this.

**Comment by Jaikant** [comment]
I would suggest to include `location` to `Fulfillment.tracking` and enhance `Location` to include `Vehicle`

**Comment by ravi-prakash-v** [comment]
@Jaikant can you elaborate further on this with an example schema?

**Comment by ravi-prakash-v** [comment]
Moving this to discussions as it is not exactly a blocking issue.
